pkg/controlsvc/status.go

```diff
@@ -17,17 +22,42 @@ func (t *statusCommandType) InitFromString(params string) (ControlCommand, error
 }
 
 func (t *statusCommandType) InitFromJSON(config map[string]interface{}) (ControlCommand, error) {
-	c := &statusCommand{}
+	fields, ok := config["requested_fields"]
+	var fieldsStr []string
+	if ok {
+		fieldsStr = make([]string, 0)
+		for _, v := range fields.([]interface{}) {
+			vStr, ok := v.(string)
+			if !ok {
+				return nil, fmt.Errorf("each element of requested_fields must be a string")
+			}
+			fieldsStr = append(fieldsStr, vStr)
+		}
+	} else {
+		fieldsStr = nil
+	}
+	c := &statusCommand{
+		fields: fieldsStr,
+	}
 	return c, nil
 }
 
 func (c *statusCommand) ControlFunc(nc *netceptor.Netceptor, cfo ControlFuncOperations) (map[string]interface{}, error) {
 	status := nc.Status()
+	precfr := make(map[string]interface{})
+	precfr["Version"] = Version
+	precfr["NodeID"] = status.NodeID
+	precfr["Connections"] = status.Connections
+	precfr["RoutingTable"] = status.RoutingTable
+	precfr["Advertisements"] = status.Advertisements
+	precfr["KnownConnectionCosts"] = status.KnownConnectionCosts
```

Comment:
This is a lot of moving around of data that maybe we're not going to use.  It probably doesn't matter that much because status probably isn't going to be called in a tight inner loop, but it would be better not to pay the runtime cost of copying a data element we know we don't need.

One way to do this would be to us a function table, like:
```
statusGetters := make(map[string]func() interface{})
statusGetters["Connections"] = func() interface{} { return status.Connections }
```

Then when assigning to cfr, you call the getter as a function, so the data copy doesn't actually happen until we know this particular item is needed.  We should also check for unknown items and ignore them rather than panic.
```
if c.fields != nil {
	for _, f := range c.fields {
		getter, ok := statusGetters[f]
		if ok {
			cfr[f] = getter()
		}
	}
}
```